### PR TITLE
WIP: python options

### PR DESCRIPTION
**Pull request by dsajdak**
I would like @aebruno comments on the python.md document before moving forward with having anyone else review.  This is a combination of our existing docs, CURC's and Compute Canada's docs though they do things a bit differently.

### CODE REVIEW

File: pages/software/python.md [hunk 1]
```diff
@@ -0,0 +1,457 @@
+# Python Options
+
+There are many options for using Python and its derivates on CCR's systems.  CCR provides, at most, two versions of Python and two versions of Anaconda as well as SciPy (Scientific Python libraries) via it's modules system.  These are updated once per year with the oldest version being rotated out.  Users will see a warning when a module is being deprecated giving you advance notice to update your workflow.
```
Comment (aebruno) [inline pages/software/python.md:3]: Given that python2 is no longer supported, we'll likely only be providing python3. So perhaps here we just say CCR provides Python3, Anaconda, and SciPy via it's module systems.

File: pages/software/python.md [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+
+CCR provides as part of the currently supported toolchains.  We do not recommend using a version included within the operating system nor the one provided in CCR's compatibility layer.  These can differ across login and compute nodes and can be replaced with newer versions, potentially affecting your project.  These are also compiled to make the best use out of the architecture you're running on rather than offering one solution for the lowest common denominator.  Rather, using one contained in CCR's software repository toolchain provides you the most flexibility and usability across all CCR systems.  We highly recommend using python virtual environments for your projects.  A python virtual environment can be created using a specific version of python and can be used in isolation of other environments.  This is convenient if your projects require different versions of Python packages or have differing dependencies.  Within the virtual environment you can install additional Python packages  These virtual environments can be stored in your home directory or within your group's project directory making them available to other members of your group.
```
Comment (aebruno) [inline pages/software/python.md:25]: It's perfectly fine to use the python from the compat layer and in our new base os images there will be no python. So we can simplify this section quite a bit, maybe something like: 

> CCR provides as python part of our software modules.  We highly recommend using python virtual environments for your projects.  A python virtual environment can be created using a specific version of python and can be used in isolation of other environments.  This is convenient if your projects require different versions of Python packages or have differing dependencies.  Within the virtual environment you can install additional Python packages  These virtual environments can be stored in your home directory or within your group's project directory making them available to other members of your group.

File: pages/software/python.md [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+
+    Properties:
+      Tools for development
+
+    You will need to load all module(s) on any one of the lines below before the "python/3.9.6" module is available to load.
+
+      gcccore/11.2.0
+
+    This module provides the following extensions:
+
```
Comment (aebruno) [inline pages/software/python.md:64]: would suggest cutting this down a bit.. no need to show the extensions here in the example module output

**Review by aebruno — CHANGES_REQUESTED** [review]
First pass at some feedback on this PR.

File: pages/software/python.md (hunk 1 above)
Comment (dsajdak) [reply pages/software/python.md]: fixed

File: pages/software/python.md (hunk 2 above)
Comment (dsajdak) [reply pages/software/python.md]: fixed

File: pages/software/python.md (hunk 3 above)
Comment (dsajdak) [reply pages/software/python.md]: removed

File: pages/software/python.md [hunk 4]
```diff
@@ -0,0 +1,457 @@
+# Python Options
+
+There are many options for using Python and its derivates on CCR's systems.  CCR provides, at most, two versions of Python and two versions of Anaconda as well as SciPy (Scientific Python libraries) via it's modules system.  These are updated once per year with the oldest version being rotated out.  Users will see a warning when a module is being deprecated giving you advance notice to update your workflow.  
+
+**This page describes the different options and we recommend you read through it all to determine which is best for your research needs**
+
+## Recommended Organization for Python
```
[the remark this answers is not in the document]
Comment (dsajdak) [reply pages/software/python.md]: modified and turned into a tip to explain our examples

File: pages/software/python.md [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+pyparsing       3.0.9
+python-dateutil 2.8.2
+setuptools      56.0.0
+six             1.16.0
+style           1.1.0
+update          0.0.1
+
+```
+
+!!! Tip
```
[the remark this answers is not in the document]
Comment (dsajdak) [reply pages/software/python.md]: didn't know this! removed

File: pages/software/python.md [hunk 6]
```diff
@@ -0,0 +1,457 @@
...
+    This module can be loaded directly: module load anaconda3/2022.05
+
+
+ccruser@vortex ~ $ module load anaconda3
+ccruser@vortex ~ $ which python
+/cvmfs/soft.ccr.buffalo.edu/versions/2023.01/easybuild/software/Core/anaconda3/2022.05/bin/python
+```
+
+#### Which environment to use?  
+
```
[the remark this answers is not in the document]
Comment (dsajdak) [reply pages/software/python.md:279]: This section was specifically about the Anaconda environments so I left it under this section.  However, I added a new section at the top as you suggested that has links to the different sections of the doc

File: pages/software/python.md [hunk 7]
```diff
@@ -0,0 +1,457 @@
...
+NOTE: In this example, we're using the suggestion directory structure as [described above](#recommended-organization-for-python)
+
+**Step 1:** If you haven't already done so, configure your `.condarc` file to [point to an alternate location](#configuring-conda-with-condarc) you have permission to write to.  Otherwise it will try creating an environment in CCR's software repository which will fail.  
+
+**Step 2:** Initialize Anaconda if you haven't already done so:  
+
+```
+ccruser@vortex ~ $ module load anaconda3
+cruser@vortex ~ $ eval "$(/cvmfs/soft.ccr.buffalo.edu/versions/2023.01/easybuild/software/Core/anaconda3/2022.05/bin/conda shell.bash hook)"
```
[the remark this answers is not in the document]
Comment (dsajdak) [reply pages/software/python.md]: fixed!

**Review by aebruno — APPROVED** [review]

**Pull request merged**